javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java

```diff
@@ -712,37 +715,100 @@ public void visit(final SuperExpr n, final Void arg) {
     @Override
     public void visit(final MethodCallExpr n, final Void arg) {
         printComment(n.getComment(), arg);
+
+        // we are at the last method call of a call chain
+        // this means we do not start reindenting for alignment or we undo it
+        AtomicBoolean lastMethodInCallChain = new AtomicBoolean(true);
+        Node node = n;
+        while (node.getParentNode()
+                .filter(NodeWithTraversableScope.class::isInstance)
+                .map(NodeWithTraversableScope.class::cast)
+                .flatMap(NodeWithTraversableScope::traverseScope)
+                .map(node::equals)
+                .orElse(false)) {
+            node = node.getParentNode().orElseThrow(AssertionError::new);
+            if (node instanceof MethodCallExpr) {
+                lastMethodInCallChain.set(false);
+                break;
+            }
+        }
+
+        // search whether there is a method call with scope in the scope already
+        // this means that we probably started reindenting for alignment there
+        AtomicBoolean methodCallWithScopeInScope = new AtomicBoolean();
+        Optional<Expression> s = n.getScope();
+        while (s.filter(NodeWithTraversableScope.class::isInstance).isPresent()) {
+            Optional<Expression> parentScope = s.map(NodeWithTraversableScope.class::cast)
+                    .flatMap(NodeWithTraversableScope::traverseScope);
+            if (s.filter(MethodCallExpr.class::isInstance).isPresent() && parentScope.isPresent()) {
+                methodCallWithScopeInScope.set(true);
+                break;
+            }
+            s = parentScope;
+        }
+
```

Comment:
Should we do all this stuff even when users usually do not use the alignment settings?

Reply:
You are probably right, I changed the order, so that `columnAlignFirstMethodChain` is calculated first and the other two are only evaluated if it is `true`.